scripts/release/common/setup.sh

```diff
@@ -7,17 +7,24 @@ if [ -z "${BUILDTIMESTAMP}" ]; then
     BUILDTIMESTAMP=$(cat "${HOME}/buildtimestamp")
     export BUILDTIMESTAMP
     echo run "${0}" with output to "${HOME}/buildlog_${BUILDTIMESTAMP}"
-    (bash "${0}" "${1}" 2>&1) | tee "${HOME}/buildlog_${BUILDTIMESTAMP}"
-    exit 0
+    bash "${0}" "${1}" 2>&1 | tee "${HOME}/buildlog_${BUILDTIMESTAMP}"
+    # http://tldp.org/LDP/abs/html/internalvariables.html#PIPESTATUSREF
+    exit "${PIPESTATUS[0]}"
```

Comment:
This will get the exit code of the first command in the pipeline, which is what we want.

Reply:
nice, but that's not what you want. You want to use `set -o pipefail` which would set the error code you get from the pipe to the first failuire.

Reply:
Ok, will change and test.

Reply:
Oh, I remember now why I used `pipestatus` instead.  I agree that in most cases setting `pipefail` is the better choice, but we do need a return value here or the script will execute one more time than we need.

``` 
if [ -z "${BUILDTIMESTAMP}" ]; then
    date "+%Y%m%d_%H%M%S" > "${HOME}/buildtimestamp"
    BUILDTIMESTAMP=$(cat "${HOME}/buildtimestamp")
    export BUILDTIMESTAMP
    echo run "${0}" with output to "${HOME}/buildlog_${BUILDTIMESTAMP}"
    bash "${0}" "${1}" 2>&1 | tee "${HOME}/buildlog_${BUILDTIMESTAMP}"
    # http://tldp.org/LDP/abs/html/internalvariables.html#PIPESTATUSREF
    exit "${PIPESTATUS[0]}"
fi
``` 
1. When the script first runs, `BUILDTIMESTAMP` isn't set so we set it and then start capturing the logs.
2. We call the script again: `bash "${0}" "${1}" 2>&1`
3. Since `BUILDTIMESTAMP` is set skip this portion and parse the file.
4. Then, we need the return value and exit the script.

Reply:
yes, setting it to `pipefail` would set the error flag once the `bash "${0}" "${1}" 2>&1 | tee "${HOME}/buildlog_${BUILDTIMESTAMP}"` command is complete. Then, given that you already have a `set -ex`, it will exit right away.

Reply:
note that pipefail would not propagate to the child bash invocation since SHELLOPTS is not exported.

Reply:
Yes, but the happy path is not to have an error.  So, if we don't explicitly return here, the script will be parsed again, which is not what we want.